### PR TITLE
5 create product

### PR DESCRIPTION
**Pull request by abeerAyad**

### CODE REVIEW

**Review by HamdyPs — CHANGES_REQUESTED** [review]
add check to prevent create same product

File: src/controllers/products/addProduct.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Response, Request } from 'express';
+import { Product } from '../../database/productSchema';
+
+const addProduct = async (req: Request, res: Response): Promise<void> => {
```
Comment (HamdyPs) [inline src/controllers/products/addProduct.ts]: add next function

File: src/controllers/products/addProduct.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  try {
+    const { title, image, rating, price, category } = req.body;
+    const isExistedProduct = await Product.findOne({ title });
+    if (isExistedProduct) {
+      res.status(400).json({ msg: 'Product Is Existing' });
```
Comment (HamdyPs) [inline src/controllers/products/addProduct.ts]: use custom error instead of sending json

File: src/controllers/products/addProduct.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    await Product.create({
+      title,
+      image,
+      rating,
+      price,
+      category,
+    });
+    res.status(201).json({ msg: 'Created Product Successfully' });
+  } catch (error) {
+    res.status(400).json({ error });
```
Comment (HamdyPs) [inline src/controllers/products/addProduct.ts]: use also custom error

**Review by HamdyPs — CHANGES_REQUESTED** [review]

**Comment by abeerAyad** [comment]
> add check to prevent create same product

done

File: src/controllers/products/addProduct.ts (hunk 1 above)
Comment (abeerAyad) [reply src/controllers/products/addProduct.ts]: done

File: src/controllers/products/addProduct.ts (hunk 2 above)
Comment (abeerAyad) [reply src/controllers/products/addProduct.ts]: done

File: src/controllers/products/addProduct.ts (hunk 3 above)
Comment (abeerAyad) [reply src/controllers/products/addProduct.ts]: done

**Review by heleena-fadi — APPROVED** [review]

**Review by HamdyPs — APPROVED** [review]

**Review by AhmedMughessib — APPROVED** [review]

**Review by AdhamHaisami — APPROVED** [review]

**Pull request merged**